From: Alexander Stein <alexander.stein@systec-electronic.com>
Date: Wed, 23 Nov 2011 16:30:13 +0100
Subject: [PATCH] Remove hardcoded cc variable

This patch is from Gentoo.
URL: http://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/sys-apps/memtest86+/files/memtest86+-4.10-hardcoded_cc.patch?revision=1.1

This patch cleans up the Makefile to allow overwrite CC and AS variable

Signed-off-by: Alexander Stein <alexander.stein@systec-electronic.com>
[mol: remove "-nopie", not available with gcc-4.6 here]
Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 Makefile |   11 +++++------
 1 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index f51b813..f8066d0 100644
--- a/Makefile
+++ b/Makefile
@@ -8,10 +8,9 @@
 #
 FDISK=/dev/fd0
 
-AS=as -32
-CC=gcc
-
-CFLAGS= -Wall -march=i486 -m32 -O2 -fomit-frame-pointer -fno-builtin -ffreestanding -fPIC -fno-stack-protector
+CFLAGS=-Wall -march=i486 -m32 -O2 -fomit-frame-pointer -fno-builtin -ffreestanding -fPIC -fno-stack-protector
+CPPFLAGS=-m32
+ASFLAGS=-32
 
 OBJS= head.o reloc.o main.o test.o init.o lib.o patn.o screen_buffer.o \
       config.o linuxbios.o memsize.o pci.o controller.o random.o spd.o \
@@ -47,10 +46,10 @@ memtest.bin: memtest_shared.bin bootsect.o setup.o memtest.bin.lds
 	memtest_shared.bin -o memtest.bin
 
 reloc.o: reloc.c
-	$(CC) -c $(CFLAGS) -fno-strict-aliasing reloc.c
+	$(CC) -c $(CFLAGS) -fno-strict-aliasing -fno-stack-protector reloc.c
 
 test.o: test.c
-	$(CC) -c -Wall -march=i486 -m32 -Os -fomit-frame-pointer -fno-builtin -ffreestanding test.c
+	$(CC) -c -Wall -march=i486 -m32 -O1 -fomit-frame-pointer -fno-builtin -ffreestanding -fno-stack-protector -fno-pie test.c
 
 clean:
 	rm -f *.o *.s *.iso memtest.bin memtest memtest_shared memtest_shared.bin
